buildSrc/src/main/groovy/org/elasticsearch/gradle/test/MultiClusterTestPlugin.groovy

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.gradle.test
+
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+
+import java.util.function.Consumer
+
+/**
+ * A plugin to add integration tests with multiple clusters
+ * Enables defining external clusters to be used in integTest
+ * using <code>clusters</code>
+ */
+public class MultiClusterTestPlugin implements Plugin<Project> {
+
+    @Override
+    public void apply(Project project) {
+        project.pluginManager.apply(StandaloneTestBasePlugin)
+        ClustersTask clustersTask = project.tasks.create('clusters', ClustersTask.class)
+        RestIntegTestTask integTest = project.tasks.create('integTest', RestIntegTestTask.class)
+        integTest.cluster.distribution = 'zip' // rest tests should run with the real zip
+        integTest.dependsOn(clustersTask)
+        clustersTask.mustRunAfter(project.precommit)
+        project.gradle.projectsEvaluated {
```

Comment:
Can you add a comment about why you need `projectsEvaluated` here? I can't tell by reading it.